### PR TITLE
Add task source name

### LINKED ISSUE #21527: Refactor `CommonScriptMsg::Task` to include the name of the task-source
Message enum is at:
 https://github.com/servo/servo/blob/da36740f0b1a48f5a4b4ed86318e110f1e63554b/components/script/script_runtime.rs#L54

Name of task sources is found at:
https://github.com/servo/servo/blob/master/components/script/task_source/mod.rs#L24

The `CommonScriptMsg::Task` should include the task-source name.

Follow up on https://github.com/servo/servo/pull/21388/

See https://github.com/servo/servo/pull/21388/commits/263ad79ba87a2b346e7f5665746719465f497af4#r212802334

### PR DESCRIPTION
**Pull request by AgustinCB**
Refactor `CommonScriptMsg::Task` to include `TaskSourceName`.

Sorry for the delay, between doing this after work and the time I spent trying to ramp up on the project, it took me a bit longer than I expected.

Test still don't pass in local, but they fail the same way on master, so I guess it's ok.

I may have forgotten something, I refactored mostly the stuff that the compiler complained about. Please let me know if I missed anything.

I tried to dump my thought process on the commit messages, so feel free to go commit by commit to understand context.

---
- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes fix #21527

- [ ] There are tests for these changes OR
- [x] These changes do not require tests because it's mostly refactor, no new behavior added.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/21583)

### CODE REVIEW

File: components/script/task_queue.rs
Comment (AgustinCB) [inline components/script/task_queue.rs]: I'm not sure this is a good idea.

I thought about refactoring this to use references instead of owned objects, but I didn't felt brave enough and went the easy way.

Any suggestion on what would be better here?

**Comment by AgustinCB** [comment]
I wasn't sure about how to (or if I should) add tests for this. If you think is appropriate, could give you give me a quick explanation on how to do it? Thanks!

File: components/script/dom/dedicatedworkerglobalscope.rs
Comment (gterzian) [inline components/script/dom/dedicatedworkerglobalscope.rs]: nit: If you insert the new code(in this case `task_source_name`) at the same level as the old code(in this case `task_category`), it makes the diff easier to understand for the reviewer(the diff will also show more clearly the difference between the old and new code, since the entire block wasn't changed in fact, see the diff for the other implementations).

In this case, since all other implementations have `task_source_name` as the first method, as does the trait signature, there is also a benefit in terms of consistency at keeping it the first(top-most) method for this implementation as well.

File: components/script/dom/dedicatedworkerglobalscope.rs
Comment (gterzian) [inline components/script/dom/dedicatedworkerglobalscope.rs]: nit: looks like you can remove this TODO

File: components/script/dom/dedicatedworkerglobalscope.rs
Comment (gterzian) [inline components/script/dom/dedicatedworkerglobalscope.rs]: Could you please add a TODO, and file a corresponding issue, pointing out that we should be using a new `unshipped-port-message-queue` task source that should be used here?

See https://html.spec.whatwg.org/multipage/#unshipped-port-message-queue (I got to this following the spec link just above the `fn PostMessage` and looking for the task source that would correspond to this specific tasks, it's often mentioned somewhere else than where the task itself is mentioned...)

File: components/script/dom/document.rs
Comment (gterzian) [inline components/script/dom/document.rs]: Please add a NOTE here that the spec mentions running these steps "in parallel", not necessarily as a queued task. For now using the `DOMManipulation` seems OK however. See Step 7 of https://fullscreen.spec.whatwg.org/#dom-element-requestfullscreen

File: components/script/dom/document.rs
Comment (gterzian) [inline components/script/dom/document.rs]: Same here, see Step 8 of https://fullscreen.spec.whatwg.org/#exit-fullscreen

File: components/script/dom/vrdisplay.rs
Comment (gterzian) [inline components/script/dom/vrdisplay.rs]: Please add a NOTE asking whether long-term we want to be using the `DOMManipulation` task source here.

File: components/script/dom/websocket.rs
Comment (gterzian) [inline components/script/dom/websocket.rs]: Please add a TODO, and file an issue, noting that we should be using a dedicated `websocket-task-source` here, see https://html.spec.whatwg.org/multipage/web-sockets.html#websocket-task-source (I got to this by first looking up this  `BufferedAmountTask`, following the spec link, and then scrolling down until I found the words "The task source for all tasks queued in this section is the WebSocket task source.")

File: components/script/dom/worklet.rs
Comment (gterzian) [inline components/script/dom/worklet.rs]: Please add a NOTE that it's unclear from the Spec which task source should be used here, see https://drafts.css-houdini.org/worklets/#dom-worklet-addmodule

**Review by gterzian — CHANGES_REQUESTED** [review]
Great work! Glad to see someone picking up this code so quickly! Just a few changes requested, and after that @jdm might want to do a final review(and the commits probably can be squashed into one after)...

**Comment by gterzian** [comment]
The build error looks unrelated https://travis-ci.org/servo/servo/jobs/423696837#L2135

By the way, the relevant test that you might want to run on your machine to see if it passes, would be 

1. `./mach test-wpt /_mozilla/mozilla/task_queue_throttling.any.worker.html`
2. and `./mach test-wpt /_mozilla/mozilla/task_queue_throttling.any.html`

File: components/script/task_queue.rs
Comment (AgustinCB) [reply components/script/task_queue.rs]: Yeah, that was my intuition here (i.e. the `entry` call was worth a clone in an small enum). Thank you for the detailed explanation!

File: components/script/task_queue.rs
Comment (gterzian) [inline components/script/task_queue.rs]: that's a nice one, there was indeed no need to unwrap the `queued_task` right inside the match...

File: components/script/task_queue.rs
Comment (gterzian) [inline components/script/task_queue.rs]: nit: I *think* this one would belong on the same line right after the `=>`, or has `rustfmt` formatted it that way?

**Review by gterzian — CHANGES_REQUESTED** [review]
One additional nit

File: components/script/task_source/user_interaction.rs
Comment (gterzian) [inline components/script/task_source/user_interaction.rs]: For all the changes inside implementation of `TaskSource`, like this one, I think you can use the associated const directly, doing something like `UserInteractionTaskSource::NAME`

**Review by gterzian — CHANGES_REQUESTED** [review]
One more set of nits :)

**Comment by AgustinCB** [comment]
Done!

**Comment by gterzian** [comment]
```

{"status": "TIMEOUT", "group": "default", "message": null, "stack": null, "subtest": null, "test": "/_webgl/conformance/context/context-release-upon-reload.html", "line": 199837, "action": "test_result", "expected": "OK"}
{"status": "TIMEOUT", "group": "default", "message": null, "stack": null, "subtest": null, "test": "/_webgl/conformance2/rendering/draw-buffers.html", "line": 199851, "action": "test_result", "expected": "OK"}
```

Only a few intermittent failure(which you can find by going to http://build.servo.org/grid and looking at the commit corresponding to what is announced at https://github.com/servo/servo/pull/21583#issuecomment-418152960)

@jdm r?

**Comment by AgustinCB** [comment]
Is there something I can do to make those tests pass?

**Comment by gterzian** [comment]
I don't think so, those are tests that will fail on occasion and that aren't related to the current changes(I think). 

We can wait for @jdm to give a, perhaps final, review...

File: components/script/task_source/networking.rs
Comment (gterzian) [inline components/script/task_source/networking.rs]: Looks like you missed one here

**Review by gterzian — CHANGES_REQUESTED** [review]
One small change...

File: components/script/dom/document.rs
Comment (gterzian) [reply components/script/dom/document.rs]: I've filed a bit more of an explanation at https://github.com/servo/servo/issues/21600

File: components/script/dom/vrdisplay.rs
Comment (gterzian) [inline components/script/dom/vrdisplay.rs]: missed this earlier, typo in the comment with `choise`...

**Review by gterzian — CHANGES_REQUESTED** [review]
One more thing that slipped my attention earlier...

**Comment by jdm** [comment]
Great work @AgustinCB, and thanks for reviewing the changes, @gterzian! I've looked them over and everything looks like what I expected :)

@bors-servo r+

**Pull request merged**
